### ISSUE TITLE
Error installing WindowsAppSDK on windows server 2022

### ISSUE THREAD

**Issue by klemensurban** [microsoft/WindowsAppSDK#2829]
Package deployment result : 0x8000000b Install failed. Please contact your software vendor.
ExtendedError: 0x8000000b The operation attempted to access data outside the valid range
ErrorMessage: Deployment Add operation with target volume C: on Package Microsoft.WindowsAppRuntime.1.1_1003.565.600.0_x64__8wekyb3d8bbwe from:  (MSIA653.tmp)  failed with error 0x8000000B. See http://go.microsoft.com/fwlink/?LinkId=235160 for help diagnosing app deployment issues.One or more install operations failed. Result: 0x8000000b

when attemting to install on windows server 2022

no support on server 2022 (2019?)? -> no WinUI 3 on Terminal Servers .. -> have to stay with WPF ?

best regards k.u.

**Comment by sachintaMSFT** [comment]
WinAppSDK framework package failing to install on Server 2022 is a known issue tracked internally by https://task.ms/40428533.

Are you seeing the issue on Sever 2019, as well ?

**Comment by klemensurban** [comment]
Hi,

i was just able to test this on windows server 2019; on this os WinAppSDK install without an error / a warning

best regards, klemens



Bundesministerium für Bildung, Wissenschaft und Forschung
Abteilung Praes/13 – IKT

DI Klemens Urban
Leiter Referat Praes/13b
+43 1 531 20-9720
Minoritenplatz 5, 1010 Wien
***@***.******@***.***>
www.bmbwf.gv.at<http://www.bmbwf.gv.at/>


Von: sachintaMSFT ***@***.***>
Gesendet: Montag, 8. August 2022 22:02
An: microsoft/WindowsAppSDK ***@***.***>
Cc: Urban Klemens ***@***.***>; Author ***@***.***>
Betreff: Re: [microsoft/WindowsAppSDK] Error installing WindowsAppSDK on windows server 2022 (Issue #2829)


WinAppSDK framework package failing to install on Server 2022 is a known issue tracked internally by https://task.ms/40428533.

Are you seeing the issue on Sever 2019, as well ?

**Comment by sachintaMSFT** [comment]
Thanks @klemensurban for testing. That is in line with our expectation that the issue is only on Windows Server 2022. As slated earlier, the fix will be in the OS itself and the fix is currently scheduled to be serviced to Server 2022 in October.

**Comment by Andrew-lll** [comment]
@sachintaMSFT @davidortinau

### Scope:
It appears that all current MAUI developers working on Server 2022 are blocked by this.  Unfortunately, waiting months until October to be able to develop apps in .net MAUI is simply not possible for our developers (which is probably the case elsewhere, and many developers _must_ run Server to emulate the prod env).  

### Workarounds?
Is there any intermediate step that can be done to workaround the issue until then: 
- a manually installable optional patch for Server 2022?
- workaround instructions to manually install WindowsAppRuntime1.1?
- a last installable version of WindowsAppRuntime? Previous versions of VS2022 preview + MAUI (VS2022 17.2?) have ran without issue on Server 2022 - is there an earlier version of the [WindowsAppRuntime](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/downloads) that we can manually roll back to, or that MAUI can roll back to in order to regain Server 2022 support?

### Prevention:
For the MAUI team, adding Server 2022 to the pre-release testing pipeline would catch these issues before they reach the customer

**Comment by michael-hawker** [comment]
This has been blocking the Windows Community Toolkit for months now in trying to update our new infrastructure for testing and now our main repository as well. We are stuck on 1.0.x until this is resolved.

**Comment by sachintaMSFT** [comment]
@Andrew-lll, 
1. There is currently no work around to the issue with installing WindowsAppSDK 1.1 on Windows Server 2022.
2. There is no WinAppSDK fix. This is an OS issue requiring an OS update.
3. We've been discussing servicing timelines and the earliest opportunity is the current October target.

**Comment by conioh** [comment]
@sachintaMSFT [wrote](https://github.com/microsoft/WindowsAppSDK/issues/2768#issuecomment-1230963089):
> It is a duplicate of the issue# https://github.com/microsoft/WindowsAppSDK/issues/2829. Latest status of the root cause and the fix is being shared there.

I'm here at https://github.com/microsoft/WindowsAppSDK/issues/2829. I don't see the root cause.
For some strange reason I don't have access to https://task.ms/40529191 so I don't know what's going there.

To be clear, "That is in line with our expectation that the issue is only on Windows Server 2022" is not a root cause nor is "As slated earlier, the fix will be in the OS itself and the fix is currently scheduled to be serviced to Server 2022 in October."

May we know what the root cause is?

**Comment by sachintaMSFT** [comment]
https://task.ms/40529191 is Microsoft internal link that is tracking the bug in OS and hence, our partners outside the company won't have access to it.

The root cause is that support for undocked DEH is incomplete in Server 2022 (support for undocked DEH was only planned for Windows 11 client Desktop and equivalent Server SKUs. It is not supposed to be in Server 2022 at all even in half complete state) and WindowsAppSDK framework package added Undocked DEH in it's 1.1 version where as it's 1.0 version doesn't have it.

**Comment by TemperedEnterprises** [comment]
I, also, cannot run any MAUI .NET app on Windows Server 2022. I don't see any fix. Windows Updates is on and functioning.

Is there a link to an install file to fix this so I can develop and run on Windows Server 2022?

**Comment by sachintaMSFT** [comment]
The fix in OS that will help WinAppSDK installer to succeed on Windows Server 2022 is anticipated to be released on 3rd Tuesday of October which is 18th October.

**Comment by Andrew-lll** [comment]
@sachintaMSFT Do you have an update on this fix (when it will be released and in which update)?

It has been almost 1 week past Oct 18th.  After testing the Oct 11 update: 2022-10 Cumulative update for 21H2 ([KB5018421](https://support.microsoft.com/en-us/topic/october-11-2022-kb5018421-os-build-20348-1129-115b1147-9568-4924-83b8-d27ab5b495be)) with the latest .NET Maui framework - this update does not resolve the issue.

**Comment by sachintaMSFT** [comment]
Did you try the one from Oct 17th - it's out of band update - [KB5020436 ](https://support.microsoft.com/en-us/topic/october-17-2022-kb5020436-os-build-20348-1131-out-of-band-18baa25a-5fc0-4118-9b89-ddbe9819e578) ?

The fix will be auto updated on Server machines on 2nd Tuesday of November. Until then, on demand update of the above KB should fix the issue.

**Comment by Andrew-lll** [comment]
@sachintaMSFT  I was able to resolve the issue on Server 2022 by installing KB5018485, which is a larger update that replaces KB5020436.  Previously, I did not install either of these as they did not indicate in the release notes that they resolved this issue - but they do!  Thank you for following up on this.

@TemperedEnterprises You can install KB5018485 as an optional update directly through windows update.  If after installing this you are still receiving errors, most likely there is a different issue at work not related to the app SDK.

**Comment by bpulliam** [comment]
Marking as Fixed since the patch is now available.  It will be in the next OS update as well.

**Comment by michael-hawker** [comment]
Looks like the KB listed above should be applied on next Tuesday's VM revision? See: https://github.com/actions/runner-images/issues/6585
